packages/snaps-simulation/src/middleware/engine.ts

```diff
@@ -26,21 +30,23 @@ export type CreateJsonRpcEngineOptions = {
  *
  * @param options - The options to use when creating the engine.
  * @param options.store - The Redux store to use.
- * @param options.hooks - Any hooks used by the middleware handlers.
+ * @param options.restrictedHooks - Any hooks used by the middleware handlers.
+ * @param options.permittedHooks - Any hooks used by the middleware handlers.
  * @param options.permissionMiddleware - The permission middleware to use.
  * @param options.endpoint - The JSON-RPC endpoint to use for Ethereum requests.
  * @returns A JSON-RPC engine.
  */
 export function createJsonRpcEngine({
   store,
-  hooks,
+  restrictedHooks,
+  permittedHooks,
   permissionMiddleware,
   endpoint = DEFAULT_JSON_RPC_ENDPOINT,
 }: CreateJsonRpcEngineOptions) {
   const engine = new JsonRpcEngine();
   engine.push(createMockMiddleware(store));
-  engine.push(createInternalMethodsMiddleware(hooks));
-  engine.push(createSnapsMethodMiddleware(true, hooks));
+  engine.push(createInternalMethodsMiddleware(restrictedHooks));
```

Comment:
I don't think these use the restricted hooks, do they? 🤔

Reply:
The `eth_accounts` implementation uses `getMnemonic`.

Reply:
Let's document it as a divergence from the client implementation then